src/main/java/team/gif/robot/commands/autoaim/LimelightAutoAim.java

```diff
@@ -28,67 +28,84 @@ public void initialize() {
-            if ( Math.abs(wheelSpeeds.leftMetersPerSecond) < velocitycap && Math.abs(wheelSpeeds.rightMetersPerSecond)< velocitycap ){
+            if (Math.abs(wheelSpeeds.leftMetersPerSecond) < velocityCap && Math.abs(wheelSpeeds.rightMetersPerSecond) < velocityCap) {
                 robotHasSettled = true;
                 System.out.println("AutoFire: Robot has settled");
             }
         }
-
-        if ( robotHasSettled ) {
+        if(robotHasSettled){
             if (targetLocked) {
-                //System.out.println(Robot.shooter.getVelocity());
-                if (Robot.shooter.getSpeed() > (targetSpeed - 20.0)) {
-
-                    // we need to check again to make sure the robot hasn't overshot the target
-                    double offset = Robot.limelight.getXOffset();
-                    if (offset > -1.0 && offset < 1.0) {
-                        Robot.indexer.setBeltMotorSpeedPercent(0.5);
-                        Robot.indexer.setMidMotorSpeed(0.4);
-                    } else {
-                        System.out.println("Offset Adjusting at: " + offset);
-                        // need to relock
-                        targetLocked = false;
-                        robotHasSettled = false;
-                    }
+                //inches
+                distanceFromHub = Math.abs((Constants.Shooter.UPPER_HUB_HEIGHT - Constants.Shooter.LIMELIGHT_HEIGHT) / Math.tan(Math.toRadians(Constants.Shooter.LIMELIGHT_ANGLE + Robot.limelight.getYOffset())));
+
+                //distance zones //more accurate than rohan (TM)
+                //Far Shot
+                if (distanceFromHub >= 200) {
+                    Robot.hood.setHoodUp();
+                    Robot.shooter.setSpeedPID(Constants.Shooter.RPM_FAR_COURT);
+                    System.out.println("distanceFromHubT: " + distanceFromHub);
+
+                    //LaunchPad
+                } else if (distanceFromHub >= 100) {
+                    Robot.hood.setHoodUp();
+                    Robot.shooter.setSpeedPID(Constants.Shooter.RPM_LAUNCHPAD);
+                    System.out.println("distanceFromHubL: " + distanceFromHub);
+
+                    //Close
+                } else if (distanceFromHub >= 50) {
+                    Robot.hood.setHoodUp();
+                    Robot.shooter.setSpeedPID(Constants.Shooter.RPM_RING_UPPER_HUB);
+                    System.out.println("distanceFromHubR: " + distanceFromHub);
+                } else {
+                    Robot.hood.setHoodDown();
+                    Robot.shooter.setSpeedPID(Constants.Shooter.RPM_FENDER_UPPER_HUB);
+                    //System.out.println("distanceFromHubF: " + distanceFromHub);
+                }
+            }
+
+            //shoot
+            if (Robot.shooter.isInTolerance() && targetLocked) {
+                Robot.indexer.setBeltMotorSpeedPercent(1.0);
+                Robot.indexer.setMidMotorSpeed(1.0);
+            }
```

Comment:
```suggestion
            }
```
```suggestion
            }
            
            double offset = Robot.limelight.getXOffset();
```

The limelight can change very quickly. Only get the XOffset once and use that in all your calculations.